packages/augur-sdk/src/api/Market.ts

```diff
@@ -0,0 +1,104 @@
+import { BigNumber } from "bignumber.js";
+import { Augur } from "../Augur";
+import { formatBytes32String } from "ethers/utils";
+import { ContractInterfaces } from "@augurproject/core";
+
+export interface CreateYesNoMarketParams {
+  endTime: BigNumber;
+  feePerCashInAttoCash: BigNumber;
+  affiliateFeeDivisor: BigNumber;
+  designatedReporter: string;
+  topic: string;
+  extraInfo: string;
+}
+
+export interface CreateCategoricalMarketParams {
+  endTime: BigNumber;
+  feePerCashInAttoCash: BigNumber;
+  affiliateFeeDivisor: BigNumber;
+  designatedReporter: string;
+  topic: string;
+  outcomes: string[];
+  extraInfo: string;
+}
+
+export interface CreateScalarMarketParams {
+  endTime: BigNumber;
+  feePerCashInAttoCash: BigNumber;
+  affiliateFeeDivisor: BigNumber;
+  designatedReporter: string;
+  topic: string;
+  extraInfo: string;
+  prices: BigNumber[];
+  numTicks: BigNumber;
+}
+
+export class Market {
+  private readonly augur: Augur;
+
+  constructor(augur: Augur) {
+    this.augur = augur;
+  }
+  
+  async createYesNoMarket(params: CreateYesNoMarketParams): Promise<ContractInterfaces.Market> {
+    const universe = this.augur.contracts.universe;
+
+    const events = await universe.createYesNoMarket(
+      params.endTime,
+      params.feePerCashInAttoCash,
+      params.affiliateFeeDivisor,
+      params.designatedReporter,
+      formatBytes32String(params.topic),
+      params.extraInfo)
+    ;
+    const marketId = this.extractMarketIdFromEvents(events);
+    return this.augur.contracts.marketFromAddress(marketId);
+  }
+
+  async createCategoricalMarket(params: CreateCategoricalMarketParams): Promise<ContractInterfaces.Market> {
+    const universe = this.augur.contracts.universe;
+
+    const events = await universe.createCategoricalMarket(
+      params.endTime,
+      params.feePerCashInAttoCash,
+      params.affiliateFeeDivisor,
+      params.designatedReporter,
+      params.outcomes,
+      formatBytes32String(params.topic),
+      params.extraInfo
```

Comment:
Should we be validating extraInfo here vs expecting a string?

Reply:
It is actually just a string. In the contract, it's purely a string that we log. We interpret it as JSON in the UI (and maybe also the middleware?).

Reply:
It is a string in the contract for event index reasons or some such efficiency thing. They would be first class params if there were room.

Reply:
I'll file another ticket to get this sorted.

Reply:
We can make it an interface then do the json serialization just before the contract call. That will break a ton of tests though so it'd be better in its own ticket.